File: Facile/The_River_II/The_River_II.py
import sys
import math

r1 = int(input())
value = "NO"

def riverSum(n):
    value = n
    for i in str(n): value += int(i)
    return value

for i in range(r1-1,1,-1):
    r2 = riverSum(i)
    if r2==r1:
        value = "YES"
        break
print(value)

File: Facile/The_River_II/The_River_II.java
import java.util.*;

class Solution {
    static int riverSum(int n) {
        String temp = Integer.toString(n);
        for (int i=0;i<temp.length();i++) {
            n += Integer.parseInt(String.valueOf(temp.charAt(i)));
        }
        return n;
    }

    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);
        int r1 = in.nextInt();
        int r2;
        String value = "NO";

        for (int i=r1-1;i>1;i--) {
            r2 = riverSum(i);
            if (r2==r1) {
                value = "YES";
                break;
            }
        }
        System.out.println(value);
        in.close();
    }
}